weather_api_q5_q6.py

```diff
@@ -0,0 +1,262 @@
+            day = forecastday.get("day", {})
+            day_dict = {
+                "Date": forecastday.get("date"),
+                "Max Temp (C)": day.get("maxtemp_c"),
+                "Min Temp (C)": day.get("mintemp_c"),
+                "Condition": day.get("condition", {}).get("text"),
+                "Max Wind (mph)": day.get("maxwind_mph"),
+                "Humidity (%)": day.get("avghumidity"),
+                "Visibility (km)": day.get("avgvis_km"),
+                "Probability of Rain (%)": day.get("daily_chance_of_rain"),
+                "Probability of Snow (%)": day.get("daily_chance_of_snow"),
+            }
+            if get_air_quality:
+                day_dict["Air Quality"] = Utils.get_air_quality(
+                    day.get("air_quality", {}).get("pm2_5")
+                )
+
+            day_dict["Hours"] = []
+
+            for hour in forecastday.get("hour", []):
+                hour_dict = {
+                    "Date/Time": Utils.format_datetime(hour.get("time")),
+                    "Temperature (C)": hour.get("temp_c"),
+                    "Condition": hour.get("condition", {}).get("text"),
+                    "RealFeel (C)": hour.get("feelslike_c"),
+                    "Wind Speed (mph)": hour.get("wind_mph"),
+                    "Humidity %": hour.get("humidity"),
+                    "Wind Pressure (in)": hour.get("pressure_in"),
+                    "Wind Gust (mph)": hour.get("gust_mph"),
+                    "Cloud Cover (%)": hour.get("cloud"),
+                    "Visibility (km)": hour.get("vis_km"),
+                    "Dew Point (C)": hour.get("dewpoint_c"),
+                    "Probability of Rain (%)": hour.get("chance_of_rain"),
+                    "Probability of Snow (%)": hour.get("chance_of_snow"),
+                }
+                if get_air_quality:
+                    hour_dict["Air Quality"] = Utils.get_air_quality(
+                        hour.get("air_quality", {}).get("pm2_5")
+                    )
+
+                day_dict["Hours"].append(hour_dict)
+
+            response["Forecast Days"].append(day_dict)
+        return response
+
+
+class Utils:
+    def get_air_quality(pm2_5: float) -> str | None:
+        """
+        A helper function to get air quality based on pm2.5 value provided.
+        """
+        if pm2_5 is None:
+            return None
+        if pm2_5 > 250:
+            return "Severe"
+        if pm2_5 > 120:
+            return "Very poor"
+        if pm2_5 > 90:
+            return "Poor"
+        if pm2_5 > 60:
+            return "Moderate"
+        if pm2_5 > 30:
+            return "Satisfactory"
+        return "Good"
+
+    def format_datetime(date_time: str) -> str:
+        """
+        Formats the datetime sent from the API.
+        """
+        if date_time:
+            date_time = datetime.strptime(date_time, "%Y-%m-%d %H:%M").strftime(
+                "%Y-%m-%d %I:%M %p"
+            )
+        return date_time
```

Comment:
I don't see any reason these functions can't just be defined as private functions for the file itself.

just adding a `_` before the function name conveys to other programmers that this is a private function meant to be used only within its own module. So there's no need for `get_air_quality` or `format_datetime` to be defined within the `Utils` class. you can get rid of the class and just define them as `_get_air_quality` and `_format_datetime` and it should be clear that these are helper functions for this file.